Lib/sndhdr.py

```diff
@@ -57,12 +58,17 @@ def what(filename):
 
 def whathdr(filename):
     """Recognize sound headers."""
-    with open(filename, 'rb') as f:
-        h = f.read(512)
-        for tf in tests:
-            res = tf(h, f)
-            if res:
-                return SndHeaders(*res)
+    try:
+        with open(filename, 'rb') as f:
+            h = f.read(512)
+            for tf in tests:
+                res = tf(h, f)
+                if res:
+                    return SndHeaders(*res)
+        return None
+
+    except (TypeError,
```

Comment:
So this TypeError is thrown when passing `None` to the `open()` call. That's specific to this function, would you prefer to just not catch it?  

```
>>> sndhdr.whathdr(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/david/Projects/PYTHON/cpython/Lib/sndhdr.py", line 62, in whathdr
    with open(filename, 'rb') as f:
TypeError: expected str, bytes or os.PathLike object, not NoneType
>>>
```

Reply:
I don't think you should catch it, from the traceback its very obvious what went wrong, since the documentation explicitly says `what()` works on files, it is expected to receive FileNotFound and invalid file name errors.

Reply:
wholly agree. I'll make the change.